Edbcloud beta (#1531)
EnterpriseDB/docs: product_docs/docs/edbcloud/beta/overview/03_security_compliance.mdx

---

> title: "Security"

Reviewer: Nit: should we remove "compliance" from the page title here, too?

Reviewer 2: @skrivov-edb can you proceed with this?

---

> EDB's cloud offering runs in your own cloud account, isolates your data from other users, and gives you control over our access to it. The key security features are:

Reviewer: This sentence feels like it could use some marketing spin - it reads like it's there because we had to put something at the top before the list. This page is essentially marketing; we should make it pop a bit.

Reviewer 3: @ba-edb We could borrow this, slightly reworked, from the pitch deck: EDB Cloud is a fully managed cloud database from the Postgres experts. The key security features are: ...
Too much?

> - **Data Isolation:** Clusters are installed and managed in your cloud environment. Complete segregation of your data is ensured: your data is never stored in a shared space, and compromise of another EDB customer's system does not put your system at risk.

Reviewer: ```suggestion
- **Data Isolation:** Clusters are installed and managed in your cloud environment. Complete segregation of your data is ensured: your data never leaves your cloud account, and compromise of another EDB Cloud customer's systems does not put your data at risk.
```

> - **Granular Access Control:** You can use _Single Sign On_ and define your own sets of _Roles_ and _Role Based Access Control_ policies to manage your individual cloud environments.

Reviewer: Two things here:

1. The use of italics seems a bit odd.
2. Can we link to Administering Your Account > Managing User Access here?

Reviewer 3: See changes in #1692

> - **Data Encryption:** Data is encrypted in motion and at rest. All network traffic is encrypted using Transport Layer Security (TLS). All data is encrypted when stored in database clusters. 

Reviewer: ```suggestion
- **Data Encryption:** All data in EDB Cloud is encrypted in motion and at rest. Network traffic is encrypted using Transport Layer Security (TLS) v1.2 or greater, where applicable, and EDB Cloud Postgres instances do not accept insecure connections. Data at rest is encrypted using AES with 256 bit keys. Data encryption keys are envelope encrypted and the wrapped data encryption keys are securely stored in an Azure Key Vault instance in your account. Encryption keys never leave your environment.
```

> - **Control Plane Logging:** Activities in the control plan related to User Roles, Organization updates, and Cluster Creation and Deletion are tracked automatically and viewed in the Activity Log.   

Reviewer: ```suggestion
- **Portal Audit Logging:** Activities in the portal, such as those related to User Roles, Organization updates, and Cluster Creation and Deletion are tracked automatically and viewed in the Activity Log.
```

> - **Database Logging and Auditing:** Creation and deletion of clusters are tracked automatically in the Activity Log and PostgreSQL logging, and auditing is supported. Functionality to track and analyze database activities is enabled automatically. The PostgreSQL Audit Extension (pgAudit) is enabled automatically for you when deploying a Postgres cluster. For EDB Postgres Advanced, the EDB Audit extension is enabled automatically for you. 

Reviewer: ```suggestion
- **Database Logging and Auditing:** Functionality to track and analyze database activities is enabled automatically. The PostgreSQL Audit Extension (pgAudit) is enabled automatically for you when deploying a Postgres cluster. For EDB Postgres Advanced, the EDB Audit extension is enabled automatically for you. 
```

Reviewer: Can we sub-bullet pgaudit and edb-audit and document how they're configured? And where the audit logs go to?

Reviewer 4: Yes, I'm just waiting for the `edb_admin` permissions merge (pg_minotiro) so that I can verify EDB Audit settings.

Reviewer 3: > I didn't see all the changes in the Database Logging and Auditing section captured in the latest version of the topic. I made a few changes that didn't seem to stick in #1692.

---

> title: "Security and Compliance"

Reviewer: We probably shouldn't even talk about compliance here yet.

Author: That is the issue of difference between  preview and GA release. We have a similar issue in UI/UX. Many screens in preview release are not up to the mark of basic UX best practices but they will be corrected in GA release. So how we tell users about what we have- do we have a lousy UI screen with list of users which you can't search? Our docs are also kind of marketing materials. I was struggling with that and decided to tell both what is available and what we are planning:. Example is here https://docs.google.com/document/d/1oPvrm5gjxlqYdejFPwvYd42wgcONNsKorZWtNIXQAhI/edit
(Please feel free to comment)

Telling user that our preview release is compliant is wrong. But not telling them that our GA release will be compliant may be also problematic.

Reviewer: > Our docs are also kind of marketing materials

The preview customers will all know that compliance is coming - it's front and center for the messaging with them. I don't think we need to worry about that as a marketing point.

Author: > However the preview customers will all know that compliance is coming - it's front and center for the messaging with them. I don't think we need to worry about that as a marketing point.

I was thinking about it. The entire section was more a marketing thing written for the right hemisphere of the brain- to make customers to *feel* comfortable with the product. May be we should remove the entire "Security and Compliance"  section form the preview release?

Reviewer 5: I would drop compliance for now, and add it back in once we have a SOC2 report, including how to get a copy etc. We should keep the Security piece, and give customers enough information to inform their own Compliance posture etc.

Author: I removed compliance related material.

---

> ### Security Governance

Reviewer: I don't think we want anything from this section.

- Cloud Security Compliance officer is responsible for maintaining compliance targets for their entire organization.

> - EDB Cloud infrastructure periodically undergoes SOC2 audit trials to ensure that EDB Cloud is always compliant with security requirements.

Reviewer: We should hold off on talking about SOC2 until our audit is complete and report is issued.

- All EDB staff undergo background checks before they are hired.
- All EDB staff are required to take mandatory security training.
- The EDB Software Development Life Cycle integrates continuous security scans of added/updated code.

### EDB Cloud Security Features

> EDB Cloud is a Database as a Service (DBaaS) which runs in your own Cloud account, isolating your data from other users & giving you control over our access to it. The key security features of EDB Cloud are as follows:

Reviewer: We don't need to describe what EDB Cloud is here. But I think we should probably have a top-level or second-level header about running in your cloud account, and what the implies.

Author: @drsm79 I can' respond to this ^^^

Author: Regarding these two paragraphs:

> EDB Cloud is a Database as a Service (DBaaS) which runs in your own Cloud account, isolating your data from other users & giving you control over our access to it. The key security features of EDB Cloud are as follows:
> - **Segregation of IT environments:** Customers database clusters are installed and managed in their own cloud environment. Complete segregation of customer's data is ensured.

I could not change that myself because I think they are not mine.  The first para was given by @drsm79  For the second para, I remember it took about half of  a day of discussion on slack with @nstott  to select proper language for this. @ba-edb you are in a better position to see the subtleties of this bullet points, please feel free to edit the language as appropriate. Thanks.

Author: ...this specific repo has liberal rules. it is OK to edit files right in github editor and make small commits right into edbcloud-beta branch. The recommended prefix for commit on the ticket is "docs(UPM-876):  ". But commit message policy is not enforced

> - **Segregation of IT environments:** Customers database clusters are installed and managed in their own cloud environment. Complete segregation of customer's data is ensured.

Reviewer: As noted above, I think this warrants a dedicated section, maybe along with Shared Responsibility.

Author: There is shared responsibility section: https://docs.google.com/spreadsheets/d/1PhKLCL9DMcvcZJVSWNMe42LUUa4WaB2-fLPFJose3O8/edit#gid=948888424

Author: The Shared Responsibility section is here https://60f76388eb1d304637d07e7b--edb-docs-staging.netlify.app/docs/edbcloud/latest/overview/04_responsibility_model/
it is compilation of your Security RACI doc https://60f76388eb1d304637d07e7b--edb-docs-staging.netlify.app/docs/edbcloud/latest/overview/04_responsibility_model/

Reviewer: Yeah, that's pretty good - after reviewing the whole thing here I think it's this bit that needs some words:

> We should have some words here about not touching the Subscriptions we're working in, eh?

- **Access Control:** EDB Cloud supports _Single Sign on_ and _Role Based Access Control_ policies. Customers may define their own sets of _Roles_ and  _Role Based Access Control_ policies to manage their individual cloud environments.
 
- **Data Encryption:** EDB Cloud encrypts customer data in motion and at rest. All network traffic is encrypted using Transport Layer Security (TLS). Data stored in database clusters is encrypted appropriately. Users can bring their own encryption keys for an additional level of control.
- **Logging and Auditing:** EDB Cloud supports Logging and Auditing
- **Backups:** EDB Cloud data is continuously replicated from the production to the hot standby data center.
- **Disaster Recovery and Business Continuity:** All components of EDB Cloud infrastructure are highly available and redundant.

> - **Incident Response:** EDB employs tools and processes which monitor the platform and has a dedicated security team as well as incident response processes.

Reviewer: How about just three bullets here: Network isolation, Encryption, and Auditing?

BC/DR and IR are both pretty deep internal things, and backups are not really security.

Author: DC/DR and IR are marketing bullet points which competitors used in their docs and customers want to hear. (I lifted the top bullets from a competitor's doc which was much longer and long winded) So, If we remove DC/DR and IR  from this section we should certainly mention them somewhere else.  Not sure that High Availability section is fit for it.  Where do we put them then?

Reviewer: I'd rather not put them anywhere. Dabblers can see that we have a SOC2, and serious customers can _get_ the SOC2.

Author: I found the doc which was the source of my inspiration for writing "Security and Compliance" section. Here it is: https://www.quickbase.com/security-and-compliance

Reviewer: Yeah... That reads like a SOC2 system description to me, honestly - it's clearly organized along standard controls critera. I'm biased towards something like [MongoDB's][1].

[1]: https://www.mongodb.com/cloud/atlas/security

Author: I have heard rumors that MongoDB owns it success to its good marketing department,  in much more degree than  to its system design.

Author: ...so when I have to lift content from a competitor,  I would rather emulate documentation that comes from a company with good marketing department than from a company with a good product. Obviously, it is even more  preferable to use some internal docs when they are available. Which bring me to my question: do we have some docs that can be useful to write this section in a better way?

Reviewer: Can you see the System Description? It has descriptions of this sort of thing if you want something to crib from.

https://docs.google.com/document/d/1BRk1Y2RQpkcuc0Zn0AKbiVErbGFEPUZo/edit#

> ### Shared Responsibility

Reviewer: We should have some words here about not touching the Subscriptions we're working in, eh?
